lib/plugins/filter/after_post_render/external_link.js

```diff
@@ -37,7 +37,7 @@ function externalLinkFilter(data) {
     config.external_link.field !== 'post') return;
 
   data.content = data.content.replace(/<a.*?(href=['"](.*?)['"]).*?>/gi, (str, hrefStr, href) => {
-    if (/target=/gi.test(str) || !isExternal(href, config)) return str;
+    if (/target=/gi.test(str) || href.startsWith('mailto:') || href.startsWith('javascript:') || !isExternal(href, config)) return str;
```

Comment:
there are countless of types (`data:`, `vbscript:` and so on), can test whether it's a link protocol (http/ftp) or not by testing whether `URL(href).origin` is null. See [`encodeURL()`](https://github.com/hexojs/hexo-util/blob/635161b3a732d3208b85269b7624e57a9816ed24/lib/encode_url.js#L24).

Reply:
OK, I will change to that.